src/Provider/Here/Tests/HereCachedResponseClient.php

```diff
@@ -0,0 +1,84 @@
+<?php
+
+/*
+ * This file is part of the Geocoder package.
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @license    MIT License
+ */
+
+namespace Geocoder\Provider\Here\Tests;
+
+use Geocoder\IntegrationTest\CachedResponseClient;
+use Http\Client\HttpClient;
+use Nyholm\Psr7\Factory\StreamFactory;
+use Nyholm\Psr7\Response;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Serve responses from local file cache.
+ *
+ * @author Sébastien Barré <sebastien@sheub.eu> override the provider-integration-tests: CachedResponseClient class
+ */
+class HereCachedResponseClient extends CachedResponseClient
```

Comment:
Good

Reply:
This is the best solution I could think of. I suggest to modify the `CachedResponseClient` only if other providers also need this functionality.